File: clients/python/Networking.py
import socket
import json

ip = "127.0.0.1"
port = 3000

BUFSIZE = 2048

username = "RhysPython"


class JsonInterpreter:
    @staticmethod
    def to_json(text):
        data = json.dumps({"text": text, "fromName": username, "commands":[]})
        return data

    @staticmethod
    def from_json(jsonData):
        message = json.loads(jsonData)
        return message

class Networking:
    def __init__(self):
        self.socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        self.socket.connect((ip, port))

    def recv(self):
        return self.socket.recv(BUFSIZE).decode()

    def sendData(self, data):
        self.socket.send((data+"\n").encode())

    def end(self):
        self.socket.close()

File: clients/ruby/networking.rb
class Networking
  class JsonInterpreter
    def toJson(text)
      data = {"text": text, "fromName": $myName, "date": DateTime.now, "commands":[]}
      return JSON.generate data
    end

    def fromJson(jsonData)
      return JSON.parse jsonData
    end
  end

  def initialize
    @socket = TCPSocket.open $ip, $port
    @json_interpreter = JsonInterpreter.new
  end

  def getSocket
    return @socket
  end

  def getJson
    return @json_interpreter
  end

  def sendData(data)
    @socket.puts data+"\n"
    @socket.flush
  end
end